src/app/api/routing.py

```diff
@@ -42,18 +45,34 @@ async def delete_entry(table: Table, entry_id: int = Path(..., gt=0)):
     return entry
 
 
-async def create_user(user_table: Table, payload: UserAuth):
+async def _create_access(login: str, password: str, scopes: str) -> AccessRead:
     # Check that username does not already exist
-    if await fetch_entry(user_table, ('username', payload.username)) is not None:
+    if await fetch_entry(access_table, [('login', login)]) is not None:
         raise HTTPException(
             status_code=400,
-            detail=f"An entry with username='{payload.username}' already exists.",
+            detail=f"An entry with login='{login}' already exists.",
         )
-
     # Hash the password
-    pwd = await security.hash_password(payload.password)
-    payload = UserCreation(username=payload.username, hashed_password=pwd, scopes=payload.scopes)
-    return await create_entry(user_table, payload)
+    pwd = await security.hash_password(password)
+    access = AccessCreation(login=login, hashed_password=pwd, scopes=scopes)
+    access_entry = AccessRead(** await create_entry(access_table, access))
+    return access_entry
+
+
+async def create_access(access_table: Table, payload: AccessAuth) -> AccessRead:
```

Comment:
If we are to avoid passing down the table, then I believe the `access_table` argument can be removed here since it's not being used

Reply:
Indeed but I wanted the create_****  method to have the same signature in the access route as in device and users routes  But we can get rid of it